01 - Weight changer (#4)
orbcollective/mpc-examples: pkg/mpc-examples/contracts/01-weight-changer/README.md

# WeightChangerController

## Summary

> WeightChangerController is a Managed Pool Controller that has the ability update token weights. 

Reviewer: ```suggestion
WeightChangerController is a Managed Pool Controller that has the ability gradually update token weights to a set of predefined weights.
```

> WeightChanger is a Managed Pool Controller that has the ability to pause swaps as well as promote rebalancing through the gradual lowering of swap fees. 

Reviewer: Controller doesn't appear to have any swap pausing functionality

## Details

> The WeightChangerController allows users update the weights of a Managed Pool's tokens over the course of 7 days.

Reviewer: ```suggestion
The WeightChangerController allows users to update the weights of a Managed Pool's tokens to one of the predefined weights sets over the course of 7 days.
```

> The WeightChanger is an extension of the NullController with added ability to update the weights of a Managed Pool's tokens over the course of 7 days.

Reviewer: Idt it's necessary to mention NullController here; sufficient just to explain the controller in isolation

## Access Control
### WeightChangerController

> The controller has no access control; All public functions can be executed by any account.

Reviewer: ```suggestion
The controller has no access control; all public functions can be executed by any account.
```

> The controller has no access control and all callable functions can be executed by any account.

Reviewer: ```suggestion
The controller has no access control; all public functions can be executed by any account.
```
